### ISSUE TITLE
#Top100 Implementation

### ISSUE THREAD

**Issue by gewure** [gewure/humansoflatecapitalism#5]
A main feature of the site.
A gallery of the top 100 memes, ratet on humansoflatecapitalism.com/index.html

the memes should be displayed like this:
a grid of 5x20 (or maybe 3x20 on mobile) memes on /top100.html with visible count of upvotes and on:hoover it shows socialbuttons(facebook,twitter,insta) to share the post (on this site, not on original source) and to upvote and downvote.

where do the original memes come from? -> index.html. we Wrapp the insta-gallery with a div that offers upvoting functionality on:hoover and whenever its upvoted, its ID(=unique url?) is persisted as Key of a Map<Url,Integer> where the values are the votes which are incremented.

From this Map<Url,Integer> the /top100.html gallery is formed.

=> we don't download the images, we just try to read the img-urls and use them (given they don't change). 

=> later we can implement a usecase "upload meme", but persisting images needs a lot of space and also produces a lot of traffic. too early for this. 

TLDR; TODO:
- Wrapping-Box with Socialbuttons and Upvotebuttons
- Firebase-Backend Map<Url,Integer> for counting votes (.. max 500 entries?)
- DOS protection of Firebase-backend (once count per cookie, max 30 counts/minute/IP)
- /top100.html Gallery-Grid

**Comment by gewure** [comment]
http://jsfiddle.net/4CqNK/8/ a pretty nice fiddle which exemples how to achieve a nice css-only overlay of the Memes in the gallery.
